r/tests/testthat/test-matrix_stats.R

```diff
@@ -187,4 +187,67 @@ test_that("svds registers generic with RSpectra", {
   # Ensure RSpectra works with BPCells and BPCells works with RSpectra
   equal_svds(ans, RSpectra::svds(i1, k=5))
   equal_svds(ans, BPCells::svds(m1, k=5))
-})
\ No newline at end of file
+})
+
+test_that("row_max and colMaxs works with negative only matrices", {
+  withr::local_seed(195123)
+  m1 <- matrix(runif(12, min=-10, max=-1), nrow = 3, ncol = 4)
+  i1 <- m1 %>% as("dgCMatrix") %>% as("IterableMatrix")
+  expect_equal(rowMaxs(i1), matrixStats::rowMaxs(m1))
+  expect_equal(colMaxs(i1), matrixStats::colMaxs(m1))
+})
+
+test_that("row_max and colMaxs works with dense matrices", {
+  withr::local_seed(195123)
+  m1_dense <- generate_dense_matrix(4, 64)
+  i1_dense <- m1_dense %>% as("dgCMatrix") %>% as("IterableMatrix")
+  expect_equal(rowMaxs(i1_dense), matrixStats::rowMaxs(m1_dense))
+  expect_equal(colMaxs(i1_dense), matrixStats::colMaxs(m1_dense))
+})
+
+test_that("row_max and colMaxs works with sparse matrices", {
+  withr::local_seed(195123)
+  m1_sparse <- generate_sparse_matrix(4, 64)
+  i1_sparse <- m1_sparse %>% as("dgCMatrix") %>% as("IterableMatrix")
+  expect_equal(rowMaxs(i1_sparse), matrixStats::rowMaxs(as.matrix(m1_sparse)))
+  expect_equal(colMaxs(i1_sparse), matrixStats::colMaxs(as.matrix(m1_sparse)))
+})
+
+test_that("row_max works for iterable matrices", {
+  withr::local_seed(195123)
+  m1_dense <- generate_dense_matrix(4, 64)
+  i1_dense <- m1_dense %>% as("dgCMatrix") %>% as("IterableMatrix")
+  expect_equal(rowMaxs(i1_dense), matrixStats::rowMaxs(as.matrix(m1_dense)))
+})
+
+
+test_that("row_max and colMaxs tranpose works", {
+  withr::local_seed(195123)
+  m1 <- generate_sparse_matrix(4, 64)
+  i1_t <- t(m1) %>% as("dgCMatrix") %>% as("IterableMatrix")
+  i2_t <- m1 %>% as("dgCMatrix") %>% as("IterableMatrix") %>% t()
+  # for the case of pre-tranposed matrics prior to conversion to IterableMatrix
+  expect_equal(rowMaxs(i1_t), matrixStats::rowMaxs(as.matrix(t(m1))))
+  expect_equal(colMaxs(i1_t), matrixStats::colMaxs(as.matrix(t(m1))))
+  # case of transposing after conversion to IterableMatrix
+  expect_equal(rowMaxs(i2_t), matrixStats::rowMaxs(as.matrix(t(m1))))
+  expect_equal(colMaxs(i2_t), matrixStats::colMaxs(as.matrix(t(m1))))
+})
+
```

Comment:
The last test case here sort of encapsulates all cases together, by iterating through every test matrix, and transpose status.  Would you prefer the ones above, or the single one below?

Reply:
My personal preference is the single combined test since it's about 3x shorter and still pretty clear to read what's going on (though either approach seems perfectly fine)

Reply:
Cool, I'll keep that in mind when making test cases in the future.  Thanks!